### PR TITLE
Feature - map dropdown contentions in /contention-classification/classifier endpoint

### PR DESCRIPTION
**Pull request by lukey-luke**
## What was the problem?

- Currently, there exists a `/contention-classification/classifier` endpoint used by VA.gov for classifying Claims For Increase. This does not, however support "NEW" type claims.

Associated tickets or Slack threads:
- https://github.com/department-of-veterans-affairs/vagov-claim-classification/issues/225

## How does this fix it?[^1]

- In an [earlier PR](https://github.com/department-of-veterans-affairs/abd-vro/pull/1950), I added an additional lookup table for mapping strings to classification values. This PR builds on top of that work..
- The `/contention-classification/classifier` endpoint preserves the existing lookup table for claims for increase.
- The dropdown lookup table will be used if a match is not found for a given Claim for increase
- The dropdown lookup table will be use if it is a "new" claim.
- The new inputs for the API are all optional or have default values in order to maintain backwards compatibility.

## How to test this PR
- navigate to `http://localhost:8120/docs` observer `/classifier` is deprecated and greyed out
- run `pytest` from `domain-cc/cc-app`
- make POST request to `http://localhost:8120/classifier`
```
{"diagnostic_code": 9999999, "claim_id": 100, "form526_submission_id": 500, "contention_text": "tuberculosis", "claim_type": "claim_for_increase"} 
```

## Rationale
- After waffling back and forth on wether new contentions warranted a new API endpoint, I decided it would be better to deal with some amount of technical debt in providing optional fields, so that an additional feature flag is not required for vets-api
- Release process:
1. release changes to abd-vro
2. API calls are still made to the same URL and API responses should follow the same behavior
3. release new changes to vets-api to allow NEW single-contention claims to route to VRO
4. API calls will be updated to include `claim_type` and `contention_text`

[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).

### CODE REVIEW

**Review by msnwatson — APPROVED** [review]
LGTM once codeql is passing

File: domain-cc/cc-app/src/python_src/api.py [hunk 1]
```diff
@@ -46,10 +46,23 @@
 
 
 @app.post("/classifier")
-def get_classification(
-    claim_for_increase: ClaimForIncrease,
-) -> Optional[PredictedClassification]:
-    classification_code = dc_lookup_table.get(claim_for_increase.diagnostic_code, None)
+def get_classification(claim: Claim) -> Optional[PredictedClassification]:
+    logging.info(
+        f"claim_id: {claim.claim_id}, form526_submission_id: {claim.form526_submission_id}"
```
[the remark this answers is not in the document]
Comment (lukey-luke) [reply domain-cc/cc-app/src/python_src/api.py:51]: these are numerical values that are non-pii

**Comment by lukey-luke** [comment]
> LGTM once codeql is passing

Resolved all codeql alerts

**Comment by lukey-luke** [comment]
Re: additional validation @tejans24 

My understanding was that the framework will automagically validate the types based on the definition of a `Claim` in [pydantic_models.py](https://github.com/department-of-veterans-affairs/abd-vro/blob/feature-dropdown-endpoint/domain-cc/cc-app/src/python_src/pydantic_models.py#L6). If it can't coerce the type, it shouldn't accept the API call.

e.g. I added `    print(f'here is the claim type d00d: {type(claim.claim_id)}')`
Screenshot 2023-08-29 at 10 19 25 AM

```
here is the claim type d00d: <class 'int'>
```

Screenshot 2023-08-29 at 10 19 38 AM


```
here is the claim type d00d: <class 'int'>
```

but this fails and throws 422
Screenshot 2023-08-29 at 10 21 34 AM
```
INFO:     127.0.0.1:55419 - "POST /classifier HTTP/1.1" 422 Unprocessable Entity
```

**Review by tejans24 — APPROVED** [review]
LGTM, thanks for providing the explanation!

**Pull request merged**